### PR TITLE
Enable arbitrary (e.g. 0) base ID

### PR DESCRIPTION
**Pull request by aalhossary**
Added a new constructor with "base" parameter.
----
In order to enable writing graphs out with IDs starting from an arbitrary number, I added this field, and parameters.

The Changes are referenced in issue: [#836](https://github.com/jgrapht/jgrapht/issues/836).

----

- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/Become-a-Contributor>
- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/How-to-make-your-first-%28code%29-contribution>
- [x] I added [unit tests](https://github.com/jgrapht/jgrapht/wiki/Unit-testing)
- [x] I added [documentation](https://github.com/jgrapht/jgrapht/wiki/How-to-write-documentation)
- [x] I followed the [Coding and Style Conventions](https://github.com/jgrapht/jgrapht/wiki/Coding-and-Style-Conventions)
- [x] I **have not** modified `HISTORY.md` or `CONTRIBUTORS.md`
- [x] I ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)

### CODE REVIEW

File: jgrapht-io/src/test/java/org/jgrapht/io/IntegerComponentNameProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package org.jgrapht.io;
```
Comment (jsichi) [inline jgrapht-io/src/test/java/org/jgrapht/io/IntegerComponentNameProviderTest.java]: Please add standard file header.

Comment (aalhossary) [reply jgrapht-io/src/test/java/org/jgrapht/io/IntegerComponentNameProviderTest.java]: Done

**Comment by aalhossary** [comment]
Added in https://github.com/jgrapht/jgrapht/pull/837/commits/f71a965a527e89a01ecf1af2c5c8dc362cc4abbf as requested.

File: jgrapht-io/src/main/java/org/jgrapht/io/IntegerComponentNameProvider.java [hunk 2]
```diff
@@ -27,20 +27,44 @@
  * @param <T> the component type
  *
  * @author Trevor Harmon
+ * @author Amr ALHOSSARY (?)
```
Comment (jsichi) [inline jgrapht-io/src/main/java/org/jgrapht/io/IntegerComponentNameProvider.java]: This is fine; you can remove the question mark.

File: jgrapht-io/src/main/java/org/jgrapht/io/IntegerComponentNameProvider.java [hunk 3]
```diff
@@ -27,20 +27,44 @@
...
 
+    /**
+     * Create a provider with the default base id (1).
+     */
+    public IntegerComponentNameProvider()
+    {
+        this(DEFAULT_BASE);
+    }
+
+    /**create a provider with a given arbitrary base
```
Comment (jsichi) [inline jgrapht-io/src/main/java/org/jgrapht/io/IntegerComponentNameProvider.java]: Capital C, and follow same formatting as previous constructor.

Comment (aalhossary) [reply jgrapht-io/src/main/java/org/jgrapht/io/IntegerComponentNameProvider.java]: Done

**Comment by aalhossary** [comment]
Am I still missing something, @jsichi?

**Comment by jsichi** [comment]
Looks good, thanks!

**Pull request merged**